sdk/log/record_test.go

```diff
@@ -230,6 +230,158 @@ func TestRecordDroppedAttributes(t *testing.T) {
 	}
 }
 
+func TestRecordAttrAllowDuplicateAttributes(t *testing.T) {
+	testcases := []struct {
+		name  string
+		attrs []log.KeyValue
+		want  []log.KeyValue
+	}{
+		{
+			name:  "EmptyKey",
+			attrs: make([]log.KeyValue, 10),
+			want:  make([]log.KeyValue, 10),
+		},
+		{
+			name: "MapKey",
+			attrs: []log.KeyValue{
+				log.Map("key", log.Int("key", 5), log.Int("key", 10)),
+			},
+			want: []log.KeyValue{
+				log.Map("key", log.Int("key", 10)),
+			},
```

Comment:
I'm unsure whether we still want to deduplicate entries inside a `log.Map` attribute.  As it stands in this PR we still deduplicate those entries even if `WithoutAttributeDeduplication` is set. 

Happy to change it though if that's the wrong approach